opentracing-mock/src/main/java/io/opentracing/mock/MockSpan.java

```diff
@@ -94,8 +95,11 @@ public void finish() {
 
     @Override
     public synchronized void finish(long finishMicros) {
-        this.finishMicros = finishMicros;
-        this.mockTracer.appendFinishedSpan(this);
+        if (!finished) {
```

Comment:
@pavolloffay "not defined" is just another way of saying "don't do it." To answer your question: in practice, high-performance Tracer impls can use free pools, and `finish()` is the cue to reuse the guts of the Span implementation.